Dockerfile

```diff
@@ -1,42 +1,82 @@
+#   nj ------  https://registry.nodejitsu.com
+#   rednpm --  http://registry.mirror.cqupt.edu.cn
+#   skimdb --  https://skimdb.npmjs.com/registry
+#   yarn ----  https://registry.yarnpkg.com
+ARG YARN_REGISTRY=https://registry.yarnpkg.com
 
-# build frontend
-WORKDIR /cloudreve_builder/Cloudreve/assets
-RUN yarn install --network-timeout 1000000
-RUN yarn run build && find . -name "*.map" -type f -delete
+# 允许通过 `docker build --build-args GOPROXY=https://goproxy.cn` 添加 go mod 代理
+ARG GOPROXY=""
 
-# build backend
-WORKDIR /cloudreve_builder/Cloudreve
-RUN zip -r - assets/build >assets.zip
-RUN tag_name=$(git describe --tags) \
-    && export COMMIT_SHA=$(git rev-parse --short HEAD) \
-    && go build -a -o cloudreve -ldflags " -X 'github.com/HFO4/cloudreve/pkg/conf.BackendVersion=$tag_name' -X 'github.com/HFO4/cloudreve/pkg/conf.LastCommit=$COMMIT_SHA'"
+# 暂不确定未来 alpine 内的 Node 版本是否影响最终编译结果, 故暂时增加打印输出
+RUN set -ex \
+    && echo "Node Version: $(node -v)" \
+    && sed -i -e "s@https://registry.yarnpkg.com@${YARN_REGISTRY}@g" yarn.lock \
+    && yarn install --network-timeout 1000000 \
+    && yarn run build \
+    && find ./build -name "*.map" -type f -delete
 
+# 编译后端代码
+WORKDIR /source
 
-# build final image
-FROM alpine:latest
+# assets.zip: 用于 go:embed 嵌入, 不可修改文件名
+RUN set -ex \
+    && zip -r - assets/build > assets.zip \
+    && go build -trimpath -o /cloudreve.bin -ldflags \
+            "-w -s \
+             -X github.com/cloudreve/Cloudreve/v3/pkg/conf.BackendVersion=$(git describe --tags) \
+             -X github.com/cloudreve/Cloudreve/v3/pkg/conf.LastCommit=$(git rev-parse --short HEAD)"
 
-WORKDIR /cloudreve
 
-RUN apk update && apk add --no-cache tzdata
+# 多阶段构建: 构建最终镜像 
+FROM alpine
 
-# we using the `Asia/Shanghai` timezone by default, you can do modification at your will
-RUN cp /usr/share/zoneinfo/Asia/Shanghai /etc/localtime \
-    && echo "Asia/Shanghai" > /etc/timezone
+# 允许通过 `docker build --build-args TIMEZONE=Africa/Abidjan` 覆盖默认时区
+ARG TZ="Asia/Shanghai"
 
-COPY --from=cloudreve_builder /cloudreve_builder/Cloudreve/cloudreve ./
+# netgo 兼容修复
+# set up nsswitch.conf for Go's "netgo" implementation
+# - https://github.com/golang/go/blob/go1.9.1/src/net/conf.go#L194-L275
+# - docker run --rm debian:stretch grep '^hosts:' /etc/nsswitch.conf
+RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
 
-# prepare permissions and aria2 dir
-RUN chmod +x ./cloudreve && mkdir -p /data/aria2 && chmod -R 766 /data/aria2
+RUN set -ex \
+    && apk add --no-cache tzdata ca-certificates bash \
+    && ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime \
+    && echo "${TZ}" > /etc/timezone
 
-EXPOSE 5212
-VOLUME ["/cloudreve/uploads", "/cloudreve/avatar", "/data"]
+# 从编译阶段镜像复制可执行文件
+COPY --from=builder /cloudreve.bin /usr/bin/cloudreve
 
-ENTRYPOINT ["./cloudreve"]
\ No newline at end of file
+# 复制启动脚本, 该脚本负责完成权限修复等预处理动作
+COPY docker-entrypoint.sh /docker-entrypoint.sh
+
+# 默认文件存储位置
+VOLUME /data
```

Comment:
previously, we using `/data` directory as the remote download (aria2) transfer station, I think you just using this directory like the data persistence directory, just a reminder, make sure there is no conflict